ee/apps/pdf-worker/src/templates/transcriptTemplate.tsx

```diff
@@ -0,0 +1,291 @@
+			{
+				name: 'moneylogo.png',
+				url: './moneylogo.png',
+			},
+		],
+	},
+	{
+		user: 'Christian Castro',
+		ts: 'Nov 21, 2022 11:00 AM',
+		msg: 'Yeah, but, can I also use some custom logo over there? I want to use this one',
+		attachments: [
+			{
+				name: 'moneylogo.png',
+				url: './moneylogo.png',
+			},
+		],
+	},
+	{
+		user: 'Christian Castro',
+		ts: 'Nov 21, 2022 11:00 AM',
+		msg: 'Yeah, but, can I also use some custom logo over there? I want to use this one',
+		attachments: [
+			{
+				name: 'moneylogo.png',
+				url: './moneylogo.png',
+			},
+		],
+	},
+];
+
+const InvoiceTableHeader = ({ values }: { values: { key: string; value: string }[] }) => (
+	<View style={styles2.container}>
+		<Text style={{ ...styles2.description, width: '60%' }}>
+			<strong>
+				<Text>{values[0].key}</Text>
+			</strong>
+			{values[0].value}
+		</Text>
+		<Text style={{ ...styles2.description, width: '40%' }}>
+			<strong>
+				<Text>{values[1].key}</Text>
+			</strong>
+			{values[1].value}
+		</Text>
+	</View>
+);
+
+// Create Document Component
+const OmnichannelTranscript = (_data: any) => (
+	<Document>
+		<Page size='A4' style={styles.page}>
+			<View style={styles.section}>
+				<div>
+					<header style={styles.header}>
+						{/* <Image style={styles.logo} src='logo.png' /> */}
+						<View style={{ flexDirection: 'column' }}>
+							<Text>Rocket.Chat</Text>
+							<Text>Chat Transcript</Text>
+						</View>
+					</header>
+				</div>
+				<div>
+					<View style={styles.tableContainer}>
+						<InvoiceTableHeader
+							values={[
+								{ key: 'Agent: ', value: 'Christian Castro' },
+								{ key: 'Date: ', value: 'Nov 21, 2022' },
+							]}
+						/>
+						<InvoiceTableHeader
+							values={[
+								{ key: 'Customer: ', value: 'Juanito Verdulero de Ponce' },
+								{ key: 'Time: ', value: '11:00 AM' },
+							]}
+						/>
+					</View>
+				</div>
+				<div>
+					<View style={styles.messagesContainer}>
+						{messages.map((message, index) => (
+							<View style={styles.message} id={`${index}`}>
+								<View style={{ flexDirection: 'row' }}>
+									<Text style={{ fontWeight: 'bold' }}>{message.user}</Text>
+									<Text style={{ marginLeft: 10, color: '#9e9e9e' }}>{message.ts}</Text>
+								</View>
+								<View style={{ marginTop: 10, flexDirection: 'column' }}>
+									<Text>{message.msg}</Text>
+								</View>
+								{/* {message.attachments?.map((attachment, index) => (
+									<View style={{ marginTop: 10, flexDirection: 'column' }} id={`attachment-${index}`}>
+										<Text style={{ color: '#9e9e9e', marginBottom: 0 }}>{attachment.name}</Text>
+										<Image style={{ width: 200 }} src={attachment.url} />
+									</View>
+								))} */}
+							</View>
+						))}
+					</View>
+				</div>
+			</View>
+		</Page>
+	</Document>
+);
+
+export default async (data: any): Promise<NodeJS.ReadableStream> => {
+	return ReactPDF.renderToStream(<OmnichannelTranscript {...{ data }} />);
+};
```

Comment:
This is the only interesting part, which is the function the service will eventually call on the template. This is the part that actually renders the file into an stream, then we consume the stream and upload the buffer :)